data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/ListToMapProcessor.java

```diff
@@ -0,0 +1,144 @@
+            } catch (IllegalArgumentException e) {
+                LOG.warn(EVENT, "Cannot find a list at the given source path [{}} on record [{}]",
+                        config.getSource(), recordEvent, e);
+                continue;
+            } catch (final Exception e) {
+                LOG.error(EVENT, "Error converting source list to map on record [{}]", recordEvent, e);
+                continue;
+            }
+
+            try {
+                updateEvent(recordEvent, targetNode);
+            } catch (final Exception e) {
+                LOG.error(EVENT, "Error updating record [{}] after converting source list to map", recordEvent, e);
+            }
+        }
+        return records;
+    }
+
+    private JsonNode getSourceNode(final Event recordEvent) {
+        final Object sourceObject = recordEvent.get(config.getSource(), Object.class);
+        return OBJECT_MAPPER.convertValue(sourceObject, JsonNode.class);
+    }
+
+    private ObjectNode constructTargetNode(JsonNode sourceNode) {
+        final ObjectNode targetNode = OBJECT_MAPPER.createObjectNode();
+        if (sourceNode.isArray()) {
+            for (final JsonNode itemNode : sourceNode) {
+                String itemKey = itemNode.get(config.getKey()).asText();
+
+                if (!config.getFlatten()) {
+                    final ArrayNode itemValueNode;
+                    if (!targetNode.has(itemKey)) {
+                        itemValueNode = OBJECT_MAPPER.createArrayNode();
+                        targetNode.set(itemKey, itemValueNode);
+                    } else {
+                        itemValueNode = (ArrayNode) targetNode.get(itemKey);
+                    }
+
+                    if (config.getValueKey() == null) {
+                        itemValueNode.add(itemNode);
+                    } else {
+                        itemValueNode.add(itemNode.get(config.getValueKey()));
+                    }
+                } else {
+                    if (!targetNode.has(itemKey) || config.getFlattenedElement() == ListToMapProcessorConfig.FlattenedElement.LAST) {
+                        if (config.getValueKey() == null) {
+                            targetNode.set(itemKey, itemNode);
+                        } else  {
+                            targetNode.set(itemKey, itemNode.get(config.getValueKey()));
+                        }
+                    }
+                }
+            }
+        } else {
+            throw new IllegalArgumentException("Cannot find a list at the given source path [{}]" + config.getSource());
+        }
+        return targetNode;
+    }
+
+    private void updateEvent(Event recordEvent, ObjectNode targetNode) {
+        final TypeReference<Map<String, Object>> mapTypeReference = new TypeReference<>() {};
+        final Map<String, Object> targetMap = OBJECT_MAPPER.convertValue(targetNode, mapTypeReference);
+
+        final boolean doWriteToRoot = Objects.isNull(config.getTarget());
```

Comment:
Just one concern, do we support JSON pointers? And what's the expected output here if my `target` is configured to empty string? I'm assuming there wouldn't be any issues.

Reply:
Will add docs in a separate PR. Thanks for the reminder! 

From what I read, we support JSON pointer partially. We do support path like `a/b/c` but not the special characters like `~0`, `~1`. They won't pass checkKey(). Empty string won't pass checkKey() as well.
https://github.com/opensearch-project/data-prepper/blob/1d0e64ddfd0c2407c8d55c621282fff23edb42ef/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/event/JacksonEvent.java#L129-L147

If `target` is an empty string, the processor won't modify the event.